### PR TITLE
refactor(function): dialog->ui

### PR DESCRIPTION
**Pull request by jayzhang**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/10283010

### CODE REVIEW

**Review by hund030 — APPROVED** [review]
Approve for function part.

File: packages/fx-core/src/plugins/resource/function/utils/depsChecker/funcPluginAdapter.ts [hunk 1]
```diff
@@ -107,34 +107,23 @@ export class FuncPluginAdapter implements IDepsAdapter {
...
-    const userSelected: string | undefined = (
-      await this._ctx.dialog.communicate(
-        new DialogMsg(DialogType.Ask, {
-          description: message,
-          type: QuestionType.Confirm,
-          options: [Messages.learnMoreButtonText, Messages.continueButtonText], // Cancel is added by default
-        })
-      )
-    ).getAnswer();
+    const res = await this._ctx.ui?.showMessage(
```
Comment (meifans) [inline packages/fx-core/src/plugins/resource/function/utils/depsChecker/funcPluginAdapter.ts:115]: Is cancel added by default?

Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/function/utils/depsChecker/funcPluginAdapter.ts:115]: Yes, if use click cancel, undefined is returned.

**Review by a1exwang — APPROVED** [review]
The environment checker part looks good to me.

**Pull request merged**